### DISCUSSION TITLE
Create Relationship via Query

### DISCUSSION THREAD

**Discussion by noor-codes** [keystonejs/keystone#7294] [Questions] [answered]
## Create Relationship via a Query

I want to create a role and assign it to a user using a query, programmatically. I don't know how to make the relationship work like shown in the picture where I can assign it to a user. 

![Screen Capture_select-area_20220221103407](https://user-images.githubusercontent.com/28804764/154898587-0e1e5597-993d-48ca-a1f8-d1e556fd9db7.png)

```javascript 
await context.db.Role.createOne({
 data: {
     name: 'admin',
     assignedTo: `How to assign the user here?`
   }
})
```

I couldn't find any docs for this specific need.

**Comment by raveling** [comment, marked as answer]
@noor-codes we can do a better job of showing how to do this in our docs. Will put that on the backlog.

Try this:


```ts
await context.db.Role.createOne({
 data: {
     name: 'admin',
     assignedTo: { connect: { id:'id-here' } }
   }
})
```

**Reply by noor-codes** [reply]
Thank you @raveling

**Reply by ghost** [reply]
@raveling 

Can you pass an array to the connect: item to connect to multiple in one query?

**Reply by dcousens** [reply]
@kennedybaird yes, you can
